src/persist-client/src/batch.rs

```diff
@@ -358,6 +358,7 @@ pub struct BatchBuilderConfig {
     pub(crate) preferred_order: RunOrder,
     pub(crate) structured_key_lower_len: usize,
     pub(crate) run_length_limit: usize,
+    pub(crate) max_runs: Option<usize>,
```

Comment:
mind adding a comment somewhere about what `None` here means? It seems like it indicates we'll maintain a single long run (the existing behavior) instead of multiple runs where we automatically compact when one gets too long?

Reply:
~Yeah, though unordered runs are treated a bit specially at the moment for backwards-compat:
- We keep the whole list of parts in memory instead of spilling runs out to S3;
- At `finish` time we put each part in an unordered run into its own single-part run.

My hope is that we can remove all the special casing once this is rolled out, but I'll definitely get a comment on here in the meantime.